File: jdi framework/jdi-uitest-core/src/main/java/com/epam/jdi/uitests/core/interfaces/common/IText.java
package com.epam.jdi.uitests.core.interfaces.common;
/*
 * Copyright 2004-2016 EPAM Systems
 *
 * This file is part of JDI project.
 *
 * JDI is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Lesser General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * JDI is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; 
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with JDI. If not, see <http://www.gnu.org/licenses/>.
 */

import com.epam.jdi.uitests.core.annotations.JDIAction;
import com.epam.jdi.uitests.core.interfaces.base.IElement;
import com.epam.jdi.uitests.core.interfaces.base.IHasValue;

/**
 * Created by Roman_Iovlev on 6/10/2015.
 */
public interface IText extends IHasValue, IElement {
    /**
     * @return Get Element’s text
     */
    @JDIAction
    String getText();

    /**
     * @param text Specify expected text
     * @return Wait while Element’s text contains expected text. Returns Element’s text
     */
    @JDIAction
    String waitText(String text);

    /**
     * @param regEx Specify expected regular expression Text
     * @return Wait while Element’s text matches regEx. Returns Element’s text
     */
    @JDIAction
    String waitMatchText(String regEx);
}

File: C#.Net/JDI/Core/Interfaces/Common/IText.cs
﻿using Epam.JDI.Core.Interfaces.Base;

namespace Epam.JDI.Core.Interfaces.Common
{
    public interface IText : IHasValue, IElement
    {
     /**
     * return: Get WebElement’s text
     */
        //TODO[JDIAction]
        string GetText { get; }

        /**
         * param: text Specify expected text
         * return: Wait while WebElement’s text contains expected text. Returns WebElement’s text
         */
        //TODO[JDIAction]
        string WaitText(string text);

        /**
         * @param regEx Specify expected regular expression Text
         * @return Wait while WebElement’s text matches regEx. Returns WebElement’s text
         */
        //TODO[JDIAction]
        string WaitMatchText(string regEx);
    }
}
